examples/cell-range-selection/src/App.js

```diff
@@ -0,0 +1,173 @@
+      border-right: 1px solid black;
+
+      :last-child {
+        border-right: 0;
+      }
+    }
+  }
+`
+
+function Table ({ columns, data }) {
+  // Use the state and functions returned from useTable to build your UI
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    // currentSelectedCellIds: cells of a current selected range
+    // selectedCellIds: All previously selected cells
+    state: { selectedCellIds, currentSelectedCellIds },
+    // getCellsBetweenId (Fn): Pass two cell Ids to get all cell Ids between them
+    getCellsBetweenId,
+    setSelectedCellIds,
+    cellsById
+  } = useTable(
+    {
+      columns,
+      data,
+      // cellIdSplitBy (string): Cell id is split by column.id + cellIdSplitBy + row.id
+      cellIdSplitBy: 'cols_rows',
+      initialState: {
+        selectedCellIds: {}
+      }
+    },
+    useCellRangeSelection
+  )
+
+  let cellsSelected = { ...currentSelectedCellIds, ...selectedCellIds }
+
+  // returns two random cell ids, this is just for the demo.
+  const getRandomCellIds = React.useCallback(() => {
+    let cloneCellIds = Object.keys(cellsById)
+    let randomCellId = () =>
+      cloneCellIds[(cloneCellIds.length * Math.random()) << 0]
+    return [randomCellId(), randomCellId()]
+  }, [cellsById])
+
+  // getCellsBetweenId returns all cell Ids between two cell Id, and then setState for selectedCellIds
+  const setCells = React.useCallback(() => {
+    const cellsBetween = getCellsBetweenId(...getRandomCellIds())
+    setSelectedCellIds(cellsBetween)
+  }, [getCellsBetweenId, setSelectedCellIds, getRandomCellIds])
+
+  return (
+    <>
+      <button onClick={setCells}>setState selectedCellIds</button>
+      <table {...getTableProps()}>
+        <thead>
+          {headerGroups.map(headerGroup => (
+            <tr {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map(column => (
+                <th {...column.getHeaderProps()}>{column.render('Header')}</th>
+              ))}
+            </tr>
+          ))}
+        </thead>
+        <tbody {...getTableBodyProps()}>
+          {rows.map((row, i) => {
+            prepareRow(row)
+            return (
+              <tr {...row.getRowProps()}>
+                {row.cells.map(cell => {
+                  return (
+                    <td
+                      {...cell.getCellRangeSelectionProps()}
+                      {...cell.getCellProps()}
+                      style={
+                        cellsSelected[cell.id]
+                          ? { backgroundColor: '#6beba8', userSelect: 'none' }
+                          : { backgroundColor: 'white', userSelect: 'none' }
+                      }
+                    >
+                      {cell.render('Cell')}
+                    </td>
+                  )
+                })}
+              </tr>
+            )
+          })}
+        </tbody>
+      </table>
+    </>
```

Comment:
Please add `selectedCellIds` state for display below table

Reply:
added